contracts/token/non-fungible/src/non_fungible.rs

```diff
@@ -0,0 +1,343 @@
+    /// * `live_until_ledger` - The ledger number at which the allowance
+    ///   expires.
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::InvalidOperator`] - If `operator` is
+    ///   `Address::ZERO`.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["approval_for_all", from: Address, operator: Address]`
+    /// * data - `[approved: bool]`
+    ///
+    /// # Notes
+    ///
+    /// We recommend using [`crate::set_approval_for_all()`] when implementing
+    /// this function.
+    fn set_approval_for_all(
+        e: &Env,
+        owner: Address,
+        operator: Address,
+        approved: bool,
+        live_until_ledger: u32,
+    );
+
+    /// Returns the account approved for `token_id` token.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `token_id` - Token id as a number.
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::NonexistentToken`] - If the token does not
+    ///   exist.
+    ///
+    /// # Notes
+    ///
+    /// We recommend using [`crate::get_approved()`] when implementing this
+    /// function.
+    fn get_approved(e: &Env, token_id: u128) -> Option<Address>;
+
+    /// Returns whether the `operator` is allowed to manage all the assets of
+    /// `owner`.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `owner` - Account of the token's owner.
+    /// * `operator` - Account to be checked.
+    ///
+    /// # Notes
+    ///
+    /// We recommend using [`crate::is_approved_for_all()`] when implementing
+    /// this function.
+    fn is_approved_for_all(e: &Env, owner: Address, operator: Address) -> bool;
+
+    /// Returns the token collection name.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    fn name(e: &Env) -> String;
+
+    /// Returns the token collection symbol.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    fn symbol(e: &Env) -> String;
+
+    /// Returns the Uniform Resource Identifier (URI) for `token_id` token.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `token_id` - Token id as a number.
+    ///
+    /// # Notes
+    ///
+    /// If the token does not exist, this function is expected to panic.
+    fn token_uri(e: &Env, token_id: u128) -> String;
+}
+
+// ################## ERRORS ##################
+
+#[contracterror]
+#[repr(u32)]
+pub enum NonFungibleTokenError {
+    /// Indicates a non-existent `token_id`.
+    NonExistentToken = 300,
+    /// Indicates an error related to the ownership over a particular token.
+    /// Used in transfers.
+    IncorrectOwner = 301,
+    /// Indicates a failure with the token `sender`. Used in transfers.
+    InvalidSender = 302,
+    /// Indicates a failure with the token `receiver`. Used in transfers.
+    InvalidReceiver = 303,
```

Comment:
Those two errors are not used yet, in what cases do we plan to use them?

Reply:
addressed in [7552cf7](https://github.com/OpenZeppelin/stellar-contracts/pull/75/commits/7552cf74f0cf98b1838cc4af1e01e3451fd69904)